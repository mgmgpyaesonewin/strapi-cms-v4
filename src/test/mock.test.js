const axios = require('axios');
const promotion = require("./promotion");

//jest.mock("axios");

describe("fetchPromotion", () => {
  describe("when API call is successful", () => {
    it("should return users list", async () => {
      let response = await promotion('http://localhost:1337/api');
      expect(response.data).toMatchSnapshot();
    });
  });

  // describe("when API call fails", () => {
  //   it("should return empty users list", async () => {
  //     // given
  //     const message = "Network Error";
  //     axios.get.mockRejectedValueOnce(new Error(message));
  //
  //     // when
  //     const result = await fetchPromotions();
  //
  //     // then
  //     expect(axios.get).toHaveBeenCalledWith(`${BASE_URL}/users`);
  //     expect(result).toEqual([]);
  //   });
  // });
});
