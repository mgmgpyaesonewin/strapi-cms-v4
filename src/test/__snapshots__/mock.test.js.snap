// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPromotion when API call is successful should return users list 1`] = `
Object {
  "responseMap": Array [
    Object {
      "action_link_android": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9tb25leXRyYW5zZmVyL3RvcHVwLz9jb250YWN0PXtudWxsfSZkaXJlY3Rpb249MjEzMTI5NjMzMSZtcF9zb3VyY2U9RGVlcGxpbms=",
      "action_link_ios": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9tb25leXRyYW5zZmVyL3RvcHVwLz9jb250YWN0PXtudWxsfSZkaXJlY3Rpb249MjEzMTI5NjMzMSZtcF9zb3VyY2U9RGVlcGxpbms=",
      "category_id": 3,
      "category_title": "Top Up",
      "external_deeplink": null,
      "hasDetails": false,
      "is_external": false,
      "photo_path": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/topup_airtime_287c1217af.png",
      "position": 1,
      "promotion_code": "code3",
      "promotion_details": null,
      "promotion_id": 2,
      "promotion_title": "Letâ€™s top up your mobile phone!",
      "promotion_type": "SPECIAL",
    },
    Object {
      "action_link_android": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9tb25leXRyYW5zZmVyL3NlbmRtb25leS8/Y29udGVudD17bnVsbH0mZGlyZWN0aW9uPTIxMzEzNjI1NzAmbXBfc291cmNlPURlZXBsaW5rJm1wX3BoX25vX3NvdXJjZT1EZWVwbGluaw==",
      "action_link_ios": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9tb25leXRyYW5zZmVyL3NlbmRtb25leS8/Y29udGVudD17bnVsbH0mZGlyZWN0aW9uPTIxMzEzNjI1NzAmbXBfc291cmNlPURlZXBsaW5rJm1wX3BoX25vX3NvdXJjZT1EZWVwbGluaw==",
      "category_id": 1,
      "category_title": "Send Money update",
      "external_deeplink": null,
      "hasDetails": false,
      "is_external": false,
      "photo_path": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/preprod_Send_Money_Prod_size_01_69d07761e3.png",
      "position": 3,
      "promotion_code": "code_2",
      "promotion_details": null,
      "promotion_id": 1,
      "promotion_title": "Free P2P Transfer",
      "promotion_type": "SPECIAL",
    },
    Object {
      "action_link_android": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9iaWxsZXIvP2RvbmF0aW9uPWZhbHNlJmJpbGxlcl9pdGVtPSU3QiUyMmFnZ3JlZ2F0b3IlMjIlM0ElMjJhZW9uJTIyJTJDJTIyYXBpSW5kaWNhdG9yJTIyJTNBMiUyQyUyMm5hbWUlMjIlM0ElMjJBRU9OJTIyJTJDJTIydHlwZSUyMiUzQSUyMk9GRkxJTkUlMjIlMkMlMjJjaGFyZ2UlMjIlM0EwLjAlMkMlMjJmZWF0dXJlTmFtZSUyMiUzQSUyMk9GRkxJTkVfQklMTF9DT0xMRUNUJTIyJTJDJTIyaWQlMjIlM0E2MzIlMkMlMjJpc0JpbGxSZWZlcmVuY2VSZXF1aXJlZCUyMiUzQWZhbHNlJTJDJTIyaXNDdXN0b21lck51bWJlclJlcXVpcmVkJTIyJTNBdHJ1ZSUyQyUyMmtleXdvcmQlMjIlM0ElMjJPRkZMSU5FX0JJTExfQ09MTEVDVCUyMiUyQyUyMm1lcmNoYW50Q29kZSUyMiUzQSUyMkFFT04lMjIlMkMlMjJvZmZzZXQlMjIlM0EwLjAlMkMlMjJwZXJjZW50YWdlJTIyJTNBMC4wJTJDJTIyc2VuZENoYXJnZSUyMiUzQTAuMCUyQyUyMmltYWdlJTIyJTNBJTIyYmlsbGVycy9hZW9uL2Flb25fbG9nby5wbmclMjIlN0QmYmlsbGVyX2NhdGVnb3J5X25hbWU9TG9hbiBSZXBheW1lbnQmYmlsbGVyX2NhdGVnb3J5X3R5cGU9MyZiaWxsZXJfaWQ9NjMyJmJpbGxlcl9jYXRlZ29yeV9pZD1PRkZMSU5FX0JJTExfQ09MTEVDVCZtcF9zb3VyY2U9RGVlcGxpbms",
      "action_link_ios": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9iaWxsZXIvP2RvbmF0aW9uPWZhbHNlJmJpbGxlcl9pdGVtPSU3QiUyMmFnZ3JlZ2F0b3IlMjIlM0ElMjJhZW9uJTIyJTJDJTIyYXBpSW5kaWNhdG9yJTIyJTNBMiUyQyUyMm5hbWUlMjIlM0ElMjJBRU9OJTIyJTJDJTIydHlwZSUyMiUzQSUyMk9GRkxJTkUlMjIlMkMlMjJjaGFyZ2UlMjIlM0EwLjAlMkMlMjJmZWF0dXJlTmFtZSUyMiUzQSUyMk9GRkxJTkVfQklMTF9DT0xMRUNUJTIyJTJDJTIyaWQlMjIlM0E2MzIlMkMlMjJpc0JpbGxSZWZlcmVuY2VSZXF1aXJlZCUyMiUzQWZhbHNlJTJDJTIyaXNDdXN0b21lck51bWJlclJlcXVpcmVkJTIyJTNBdHJ1ZSUyQyUyMmtleXdvcmQlMjIlM0ElMjJPRkZMSU5FX0JJTExfQ09MTEVDVCUyMiUyQyUyMm1lcmNoYW50Q29kZSUyMiUzQSUyMkFFT04lMjIlMkMlMjJvZmZzZXQlMjIlM0EwLjAlMkMlMjJwZXJjZW50YWdlJTIyJTNBMC4wJTJDJTIyc2VuZENoYXJnZSUyMiUzQTAuMCUyQyUyMmltYWdlJTIyJTNBJTIyYmlsbGVycy9hZW9uL2Flb25fbG9nby5wbmclMjIlN0QmYmlsbGVyX2NhdGVnb3J5X25hbWU9TG9hbiBSZXBheW1lbnQmYmlsbGVyX2NhdGVnb3J5X3R5cGU9MyZiaWxsZXJfaWQ9NjMyJmJpbGxlcl9jYXRlZ29yeV9pZD1PRkZMSU5FX0JJTExfQ09MTEVDVCZtcF9zb3VyY2U9RGVlcGxpbms=",
      "category_id": 4,
      "category_title": "Loan",
      "external_deeplink": null,
      "hasDetails": false,
      "is_external": false,
      "photo_path": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/AEON_PROD_a583ce0ccb.png",
      "position": 2,
      "promotion_code": "code_1",
      "promotion_details": null,
      "promotion_id": 3,
      "promotion_title": "AEON Free Transaction",
      "promotion_type": "NORMAL",
    },
    Object {
      "action_link_android": "d2F2ZXBheTovL2hvbWUvdWkvdmlldy9iaWxsZXIvP2RvbmF0aW9uPWZhbHNlJmJpbGxlcl9pdGVtPSU3QiUyMmFnZ3JlZ2F0b3IlMjIlM0ElMjJhZW9uJTIyJTJDJTIyYXBpSW5kaWNhdG9yJTIyJTNBMiUyQyUyMm5hbWUlMjIlM0ElMjJBRU9OJTIyJTJDJTIydHlwZSUyMiUzQSUyMk9GRkxJTkUlMjIlMkMlMjJjaGFyZ2UlMjIlM0EwLjAlMkMlMjJmZWF0dXJlTmFtZSUyMiUzQSUyMk9GRkxJTkVfQklMTF9DT0xMRUNUJTIyJTJDJTIyaWQlMjIlM0E2MzIlMkMlMjJpc0JpbGxSZWZlcmVuY2VSZXF1aXJlZCUyMiUzQWZhbHNlJTJDJTIyaXNDdXN0b21lck51bWJlclJlcXVpcmVkJTIyJTNBdHJ1ZSUyQyUyMmtleXdvcmQlMjIlM0ElMjJPRkZMSU5FX0JJTExfQ09MTEVDVCUyMiUyQyUyMm1lcmNoYW50Q29kZSUyMiUzQSUyMkFFT04lMjIlMkMlMjJvZmZzZXQlMjIlM0EwLjAlMkMlMjJwZXJjZW50YWdlJTIyJTNBMC4wJTJDJTIyc2VuZENoYXJnZSUyMiUzQTAuMCUyQyUyMmltYWdlJTIyJTNBJTIyYmlsbGVycy9hZW9uL2Flb25fbG9nby5wbmclMjIlN0QmYmlsbGVyX2NhdGVnb3J5X25hbWU9TG9hbiBSZXBheW1lbnQmYmlsbGVyX2NhdGVnb3J5X3R5cGU9MyZiaWxsZXJfaWQ9NjMyJmJpbGxlcl9jYXRlZ29yeV9pZD1PRkZMSU5FX0JJTExfQ09MTEVDVCZtcF9zb3VyY2U9RGVlcGxpbms",
      "action_link_ios": null,
      "category_id": 5,
      "category_title": "Biller",
      "external_deeplink": null,
      "hasDetails": false,
      "is_external": false,
      "photo_path": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/promotion_square_sticker_sign_banner_260nw_1427755229_cc6715c85a.webp",
      "position": 5,
      "promotion_code": "gdgdrgdr",
      "promotion_details": null,
      "promotion_id": 4,
      "promotion_title": "Biller Test",
      "promotion_type": "NORMAL",
    },
  ],
  "status": "Success",
}
`;
